### PR TITLE
Init Timeline in Stopping state in create_timeline_struct when Cause::Delete

### PR DESCRIPTION
**Pull request by LizardWizzard**
See https://github.com/neondatabase/neon/pull/4552#discussion_r1258368127 for context.

TLDR: use CreateTimelineCause to infer desired state instead of using .set_stopping after initialization

### CODE REVIEW

**Review by koivunej — APPROVED** [review]
I wonder if this is a bit sneaky but regardless, seems valid.

**Comment by LizardWizzard** [comment]
> I wonder if this is a bit sneaky but regardless, seems valid.

Yeah, that was @problame idea. I'm ok with either option, but original one looks a bit better to me, because its explicit

File: pageserver/src/tenant.rs [hunk 1]
```diff
@@ -2031,7 +2032,13 @@ impl Tenant {
             pg_version,
             initial_logical_size_can_start.cloned(),
             initial_logical_size_attempt.cloned(),
-        ))
+        );
+
+        if matches!(cause, CreateTimelineCause::Delete) {
+            timeline.set_state(TimelineState::Stopping);
+        }
```
Comment (problame) [inline pageserver/src/tenant.rs:2039]: Why not push this down further into `Timeline::new` ?

**Review by problame — APPROVED** [review]
Please adapt PR description & commit message to repro rationale from the original review comment.

File: pageserver/src/tenant.rs (hunk 1 above)
Comment (LizardWizzard) [reply pageserver/src/tenant.rs]: See updated version

**Review by koivunej — APPROVED** [review]

File: pageserver/src/tenant.rs (hunk 1 above)
Comment (LizardWizzard) [reply pageserver/src/tenant.rs]: Updated version got reviewed, resolving

**Pull request merged**